test/functional/neutronless/l2adjacent/test_disconnected_service_creation.py

```diff
@@ -0,0 +1,257 @@
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(SEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS | SEG_DEPENDENT_LB_URIS
+
+
+def test_nosegid_lb(setup_neutronless_test, configure_icd, bigip):
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    icontroldriver._common_service_handler(NOSEGID_CREATELB)
+    after_create_registry = register_device(bigip)
+    new_uris = set(after_create_registry.keys()) - set(start_registry.keys())
+    assert new_uris == SEG_INDEPENDENT_LB_URIS
+
+
+@pytest.mark.skip(reason="This test will not pass until triggering the'\
+    ' timeout condition logs a message with the appropriate literal string.'\
+    ' Perhaps this will occur when the short-circuit-for-segless bug is'\
+    ' fixed.")
+def test_nosegid_lb_timeout(setup_neutronless_test, configure_icd, bigip):
+    # Configure
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    gtimeout = icontroldriver.conf.f5_network_segment_gross_timeout
+    poll_interval = icontroldriver.conf.f5_network_segment_polling_interval
+    # Set timers
+    start_time = time.time()
+    timeout = start_time + gtimeout
+    # Configure logging
+    logger.propagate = False
+    logger.removeHandler(logger.handlers[0])
+    logger.setLevel(logging.ERROR)
+    logfilename = '/root/devenv/f5-openstack-agent/test/functional/'\
+                  'neutronless/l2adjacent/logtimeout.txt'
+    timeoutfh = logging.FileHandler(logfilename)
+    timeoutfh.setLevel(logging.ERROR)
+    logger.addHandler(timeoutfh)
+    # Begin operations
+    while time.time() < (timeout + (2*poll_interval)):
+        time.sleep(poll_interval)
+        icontroldriver._common_service_handler(NOSEGID_CREATELB)
+        create_registry = register_device(bigip)
+        new_uris = set(create_registry.keys()) - set(start_registry.keys())
+        assert new_uris == SEG_INDEPENDENT_LB_URIS
+    timeoutfh.close()
+    logger.removeHandler(logger.handlers[0])
+    assert "TIMEOUT: failed to connect " in open(logfilename).read()
+
+
+def test_nosegid_to_segid(setup_neutronless_test, configure_icd, bigip):
+    # Configure
+    start_registry = register_device(bigip)
+    icontroldriver = configure_icd(FEATURE_ON)
+    rdscache = icontroldriver.network_builder.rds_cache
+    logger.info(rdscache)
+    gtimeout = icontroldriver.conf.f5_network_segment_gross_timeout
+    poll_interval = icontroldriver.conf.f5_network_segment_polling_interval
+    # Set timers
+    start_time = time.time()
+    timeout = start_time + gtimeout
+    # Begin operations
+    while time.time() < (timeout - (2*poll_interval)):
+        time.sleep(poll_interval)
+        icontroldriver._common_service_handler(NOSEGID_CREATELB)
```

Comment:
I don't think this really tests the feature.  You need to create the lb and listener with no segid, and sit in the while loop with NOSEGID_CREATELISTENER.